File: services/owasp-seeder/tests/fixtures/vulnerable_code/python/command_injection.py
"""
Базовый пример: Command Injection уязвимость

Функции содержат критические уязвимости command injection через использование
shell=True в subprocess и небезопасной обработки пользовательского ввода.
"""

import subprocess
import os

def ping_system(target_ip):
  
    command = f"ping -c 4 {target_ip}"
    result = subprocess.run(command, shell=True, capture_output=True)

    # Безопасная альтернатива:
    # result = subprocess.run(["ping", "-c", "4", target_ip], capture_output=True)

    return result.stdout.decode()

def execute_system_command(command):

    os.system(command)

    # Безопасная альтернатива:
    # subprocess.run(command.split(), capture_output=True)

def grep_in_files(pattern, filename):

    command = f"grep '{pattern}' {filename}"
    result = subprocess.run(command, shell=True, capture_output=True)

    # Безопасная альтернатива:
    # with open(filename, 'r') as f:
    #     return [line for line in f if pattern in line]

    return result.stdout.decode()

File: services/owasp-seeder/tests/fixtures/vulnerable_code/go/command_injection.go
package main

import (
	"fmt"
	"os/exec"
	"strings"
)

func PingSystem(targetIP string) error {
	cmd := exec.Command("ping", "-c", "4", targetIP)
	output, err := cmd.CombinedOutput()
	if err != nil {
		return err
	}

	fmt.Println(string(output))

	return nil
}

func GrepInFiles(pattern, filename string) (string, error) {
	cmd := exec.Command("sh", "-c", fmt.Sprintf("grep '%s' %s", pattern, filename))
	output, err := cmd.CombinedOutput()
	if err != nil {
		return "", err
	}

	return string(output), nil

}

func ExecuteUserCommand(userInput string) error {
	args := strings.Fields(userInput)
	cmd := exec.Command(args[0], args[1:]...)

	output, err := cmd.CombinedOutput()
	if err != nil {
		return err
	}

	fmt.Println(string(output))
	return nil

}
